lib/tokenize.h

```diff
@@ -829,7 +829,7 @@ class CPPCHECKLIB Tokenizer {
     void setVarIdClassDeclaration(const Token * const startToken,
                                   const VariableMap &variableMap,
                                   const nonneg int scopeStartVarId,
-                                  std::map<int, std::map<std::string,int>>& structMembers);
+                                  std::map<int, std::map<std::string,int>>& structMembers) const;
```

Comment:
I am not sure if this should be const. The name "set.." indicates it has some side effect? If it only modifies `structMembers` then I think it would be fine that it is const.

Reply:
There are also `setVarIdClassDeclaration` and `setVarIdClassFunction`, which are const already.

Reply:
hmm maybe the check is too pedantic.. this is why we write: ` "technically" the function can be const.`

but it seems unfortunate to allow this noise.

it does not lead to bugs to have a non-const method as far as I know. but if you make it "const" and think it's safe to call it then there can be bugs.

Reply:
I meant to write "`setVarIdStructMembers` and `setVarIdClassFunction`" above.
I think it's just good practice to make everything const that can be const, since it tells you something about what the function does. Although sometimes the compiler's idea of const is not aligned with the developer's (e.g. an object owns a pointer to a buffer, but the elements of the buffer can be modified without violating const rules)

Reply:
As far as I see the tokenlist is modified by this method so it's not sensible to make this const.

Reply:
So should const be removed from setVarIdStructMembers and setVarIdClassFunction also for consistency?

Reply:
yes.